src/gpu/sycl/ref_eltwise.hpp

```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#ifndef GPU_SYCL_REF_ELTWISE_HPP
+#define GPU_SYCL_REF_ELTWISE_HPP
+
+#include "gpu/gpu_eltwise_pd.hpp"
+#include "gpu/sycl/sycl_gpu_primitive.hpp"
+#include "gpu/sycl/sycl_io_helper.hpp"
+#include "gpu/sycl/sycl_post_ops.hpp"
+#include "gpu/sycl/sycl_primitive_conf.hpp"
+#include "gpu/sycl/sycl_q10n.hpp"
+#include "gpu/sycl/sycl_types.hpp"
+#include "sycl/sycl_stream.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace sycl {
+
+template <impl::data_type_t data_type>
```

Comment:
The templatization on datatype seems unnecessary here, as it is only used for checking datatypes of arguments.

Could you please remove this templatization as it will save library size, and dispatch time?

Reply:
We changed the code as per comment. Currently removed the template class for FWD & BWD.